main.tf

```diff
@@ -144,6 +144,30 @@ resource "github_repository" "repository" {
   }
 }
 
+# ---------------------------------------------------------------------------------------------------------------------
+# Manage branches
+# https://registry.terraform.io/providers/integrations/github/latest/docs/resources/branch
+# ---------------------------------------------------------------------------------------------------------------------
+
+locals {
+  branches_temp = [
+    for b in var.branches : try({ name = tostring(b) }, b)
+  ]
+
+  branches = {
+    for b in local.branches_temp : lookup(b, "id", b.name) => b
```

Comment:
Hmm is it OK that we don't document the `id` attribute ? The module caller could pass objects with `id` instead of `name` and it would work but that is undocumented.

Reply:
fair point! I just changed that :)